src/main/java/com/nutomic/syncthingandroid/util/AsyncState.java

```diff
@@ -0,0 +1,30 @@
+package com.nutomic.syncthingandroid.util;
+
+import android.os.AsyncTask;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * Tracks the current state of a running AsyncTask without keeping a reference to it.
+ * This allows the object to be GCed as soon as it is finished. Releasing the resources
+ * allocated by the object.
+ */
+public class AsyncState {

```

Comment:
Regarding the "premature optimization". It's no problem to replace the WeakReference by a real object reference. It just didn't feel right to return the full AsyncTask because this would allow the caller to cancel the operation or interfere with the inner workings of getEvents in some way. So I wanted an interface class that hides the inner workings of getEvents and just allows to poll the state of the operation. So I reused what I had lying around (I wrote this for an other project that used much more worker threads).